### DISCUSSION TITLE
Inaccurate FDPM amplitude and phase derivative values with respect to mu_a and mu_s'

### DISCUSSION THREAD

**Discussion by lmalenfant** [VirtualPhotonics/Vts.Gui.Wpf#46] [General]
I am creating a discussion for this issue because as I work though it, it is becoming apparent that we will not be able to achieve the result we want from the GUI alone, I think we will need to open up functionality in the VTS.

I did implement the solution I thought might be possible by calling ComputeReflectance twice but this does not work because it does more than just calculate the derivatives.

@hayakawa16 & @dcuccia I would love your input on this, I think we can use some of what I added in the GUI with new functionality in the VTS to get the desired result.

**Comment by hayakawa16** [comment]
I created a mock up that might replace the existing radio buttons under "Model/Analysis Output".  For all cases, even those which have checked "use spectral panel inputs" and/or "allow multi-axis selection", the radio button toggle setup shown is based on if the x-axis of the plot is "ft" (Complex results) or not (Real results).  My thought is that for Real plots, "R" would be default selection with option to also toggle on any of the derivatives (first rectangle below).  For Complex plots, "Complex" would be the default selection with option to also toggle on any of the derivatives (second rectangle).  If the user selects "Phase" or "Amp", then none of the derivative toggles can be selected or one of them can be selected and the dPhase/dX or dAmp/dX would be plotted.  If a derivative toggle is selected, the option selected in the top row would remain on.

Please feel free to throw darts.

![model_analysis_output_mock](https://github.com/VirtualPhotonics/Vts.Gui.Wpf/assets/26939347/cea4f4b6-be8d-4636-b468-bf9137d7114a)

**Reply by lmalenfant** [reply]
This layout will not work as is because the radio buttons would have to stay inline and if we removed R when we click a derivative, there would be no way to get back to R.

Here are 2 screenshots of the VTS with the changes you proposed, would this work? 
![image](https://github.com/VirtualPhotonics/Vts.Gui.Wpf/assets/8517009/3865c71e-2d4c-482f-8e9b-8fabf061f080)

![image](https://github.com/VirtualPhotonics/Vts.Gui.Wpf/assets/8517009/0aaa664f-d85f-40d4-a75a-14bf16cbdc98)

**Comment by lmalenfant** [comment]
With @hayakawa16's changes and some modifications to the GUI, we can now create the phase and amplitude plots:

Phase
![image](https://github.com/VirtualPhotonics/Vts.Gui.Wpf/assets/8517009/64ef2e92-d0bd-4a0f-8ea9-d6f410ad3c1f)

Amplitude
![image](https://github.com/VirtualPhotonics/Vts.Gui.Wpf/assets/8517009/9eb4155c-d27d-47d8-b384-4c50498754da)

We added the Rho values to the plot data and changed the names of the derivatives, we removed the plot toggle on the plot tab.

**Comment by hayakawa16** [comment]
This looks good @lmalenfant!

**Comment by lmalenfant** [comment]
Closing this discussion because this was resolved in issue #42
